react/ProductList.tsx

```diff
@@ -111,8 +112,37 @@ const ProductGroup: StorefrontFunctionComponent<Props> = props => {
   )
 }
 
+type TotalItemsType =
+  | 'total'
+  | 'distinct'
+  | 'totalAvailable'
+  | 'distinctAvailable'
+
+const countModeHandle = (countMode: TotalItemsType, arr: Item[]) => {
+  const options = {
+    distinctAvailable: arr.reduce((itemQuantity: number, item: Item) => {
+      if (item.availability === 'available') {
+        return ++itemQuantity
+      }
+      return itemQuantity
+    }, 0),
+    totalAvailable: arr.reduce((itemQuantity: number, item: Item) => {
+      if (item.availability === 'available') {
+        return itemQuantity + item.quantity
+      }
+      return itemQuantity
+    }, 0),
+    total: arr.reduce((itemQuantity: number, item: Item) => {
+      return itemQuantity + item.quantity
+    }, 0),
+    distinct: arr.length,
+  }
+
+  return options[countMode] ?? options.distinct
+}
```

Comment:
While this function works as expected, it unnecessarily loops the `arr` multiple times for any count mode. Can you refactor to multiple `if(countMode === '...') { return arr.reduce(...)} `?